pkg/middleware/validation.go

```diff
@@ -45,31 +49,50 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(context context.Context, message string, mod string, version string) {
+	if len(message) > 0 {
+		entry := log.EntryFromContext(context)
+		entry.Warnf("error validating %s@%s %s", mod, version, message)
```

Comment:
I am not very familiar with the codebase yet, but does it not make more sense to log this at `info` level? usually warning means that this can be potential harmfull forthe application 🤔

Reply:
I think a warning is sufficient given that if the component was allowed to be retrieved it could have harmful implications to the client, i.e. it is not a valid package (for whatever reason)

Reply:
make sense now :)